### PR TITLE
DP-17194 add data tab metatags to service details, info details and curated lists

### PR DESCRIPTION
**Pull request by a-username**
**Description:**
Added 2 new tokens to `mass_metatag.module`. The first, `mass_related_terms`, gets the field off the taxonomy term associated with an info details or service details page depending on the data type selected. The second, `mass_services_category`, moves some of the the code from `mass_theme.theme` that set the value of the `category` tag and implements a check to see whether the `field_data_flag` field is selected to determine if category should be `data` or `services`.

Additionally, all remaining `category` tags previously defined in `mass_theme.theme` have been declared statically in metatag config.

Unit tests for checking the appropriate tags are set when conditions for the data type fields are met have been added as well.

**Jira:**
https://jira.mass.gov/browse/DP-17194

**To Test:**
Service details + Info details
- [ ] Edit a service details or info details page.
- [ ] Check the box to "Include this page in the “Data” tab within search" under Data flag
- [ ] Under "Data type" select "Dataset"
- [ ] Save the page and view the node
- [ ] Inspect the node and open the `<head>` tag
- [ ] Find the `mg_type` and `category` tags
- [ ] Verify `mg_type` is set to "dataset" and `category` is set to "data"
- [ ] Re-edit the node and set "Data type" to "Data catalog"
- [ ] Save and view the node
- [ ] Verify `mg_type` is now set to "data-catalog" in the inspector
- [ ] Re-edit the node and set "Data type" to "Data resource"
- [ ] Under "Data resource type" select any of the options
- [ ] Save and view the node
- [ ] Verify `mg_type` corresponds to the options selected for "Data resource type"

Curated lists
- [ ] Edit a curated list page
- [ ] Check the box to "Include this page in the “Data” tab within search" under Data flag
- [ ] Save the page and view the node
- [ ] Inspect the node and open the `<head>` tag
- [ ] Find the `category` tag
- [ ] Verify `category` is set to "data"

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by rbayliss** [comment]
Just wanted to pop in to say this looks like an excellent cleanup!

File: docroot/modules/custom/mass_metatag/mass_metatag.module [hunk 1]
```diff
@@ -195,6 +205,39 @@ function mass_metatag_tokens($type, $tokens, array $data, array $options, Bubble
...
           break;
+
+        case 'mass_related_terms':
+          $tags = [];
+          if ($data['node']->hasField('field_details_data_type') && $data_type = $data['node']->get('field_details_data_type')) {
+            $items = $data_type->referencedEntities();
+            if (!empty($items)) {
+              $type_name = \Drupal::service('entity_type.manager')
+                ->getStorage('taxonomy_term')
+                ->load($data_type->getValue()[0]['target_id'])
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/mass_metatag.module]: It looks to me like you already have a loaded entity in $items[0]->get('name')->getString(). In general, the getString() method on fields can avoid code like `getValue()[0]['value']` which happens frequently here.

File: docroot/modules/custom/mass_metatag/mass_metatag.module [hunk 2]
```diff
@@ -195,6 +205,39 @@ function mass_metatag_tokens($type, $tokens, array $data, array $options, Bubble
...
+                ->get('name')->getValue()[0]['value'];
+              $data_resources = $data['node']->get('field_data_resource_type')->referencedEntities();
+              if (!empty($data_resources) && strpos($type_name, 'resource') !== FALSE) {
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/mass_metatag.module]: Lets use the FieldItem->isEmpty() method in various places. So, `$data['node']->get('field_data_resource_type')->isEmpty()`

File: docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CuratedListDataTabMetadataTest.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      'moderation_state' => 'published',
+    ]);
+    return $node;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExpectedMetatags(ContentEntityInterface $entity) {
+    return array_merge(parent::getExpectedMetatags($entity), [
```
Comment (weitzman) [inline docroot/modules/custom/mass_metatag/tests/src/ExistingSite/CuratedListDataTabMetadataTest.php:39]: Great tests. Thanks!

**Review by weitzman — CHANGES_REQUESTED** [review]
Just a little Field API tidying before approval ... Note that I resolved a conflict so please pull before you start working again.

**Comment by a-username** [comment]
@weitzman I think I've addressed your comments from above. Mind taking another quick look?

**Pull request merged**